**Class Notes**

- Parsimony
    - IDs the simplest explanation for the data
    - assumes observed char diff. resulted from the fewest poss. mutations
    - seeks the three that yeilds the lowest possible **parsimony score** -- the sum of cost of all mutations found in the tree
    - doesn't use the genetic distance matrix at all

    - algorithm uses the n x m alignment matrix
        - n = num species
        - m = # characters
    - GOAL: determine what character strings at internal nodes would best explain the char strings for the n observed species
    - "ancestral state reconstruction"

    - input: Tree T with ea. leaf labeled by an m-char string
    - output: labeling of internal vertices of the tree T minimizing the parsimony score

- Having a stop codon right before reading genes for getting wings prevents wings in animal

- Phylogeny options
    - Naive clustering (UPGMA)
        - bad due to ultrametric (fixed evo rate)

    - Parsimony
        - fails for long distances w/ multiple changes in a single site; works well for short/dense trees
    
    - max likelihood
    ![Max Likelihood Tree](./images/10-25-2023/max%20likelihood%20tree.png)  

    **NOTE**: The probability of going from somewhere like 0 to 6 along v6 and soforth -- pretty markovian

        - slow; RaxML is a fast approx.

    - neighbor-joining
        - finds a pair of leaves that're close to ea. other, but far from other leaves: implicitly finds a pair of neighboring leaves
        - advantages: works well for additive and other non-additive matrices
        - not max likelihood; usually close or same as max likelihood, and fast